bin.js

```diff
@@ -31,3 +34,12 @@ function getConfig() {
   }
 }
 
+function getRules(set) {
+  var configFileName = set ? ('eslint-config-' + set) : null
```

Comment:
>>Installing eslint-find-new-rules globally and would not work, as referenced configs couldn't be found.

>I'm not a fan of installing tools like this globally anyway, so that's fine.

Ok, thought similar about this. I just wanted to be sure, we make the same assumptions here.

---

I thought of achieving the following:

1. Determine a particular eslint config als base
  1. That is, when running `eslint-find-new-rules` without any argument: taking the currently applicable eslint config (from an `.eslintrc` file — or `.js`, `.json`, `yml` — or its corresponding property within the project's `package.json`).
  2. When running `eslint-find-new-rules` with an argument: using that as particular eslint config file (even the corresponding property within a package.json, when specified as argument).
2. Collect all rules contained within the determined eslint config, as well as those from configs referenced to via `extends`.
3. Do **not** take any rules into account that may be applicable by considering the current location on the filesystem.

According to your example project:

- When testing against `index.js`: collect only the rules within.
- When testing against `.eslintrc`: collect all the rules within, extending the rules of `react.js`, extending the rules of `index.js`.

When we use the eslint api's method and there would be the sideeffect of accidentally taking environmental configs into account, that would also be the case when copying the file to test anywhere else on the filesystem (but within / somewhere below the same project root), because the api's method would look up all the way up to the project root ... at least this is as I currently understand its working behaviour. Or am I missing something?

My time's running out currently, and I have another appointment soon, but I'll have a closer look and run some more tests later tonight and tomorrow.